papermill/abs.py

```diff
@@ -32,13 +34,24 @@ def _split_url(self, url):
 
     def read(self, url):
         params = self._split_url(url)
+        output_stream = io.BytesIO()
         block_blob_service = self._block_blob_service(
             account_name=params["account"], sas_token=params["sas_token"]
         )
-        return block_blob_service.get_blob_to_text(
-            container_name=params["container"], blob_name=params["blob"]
+
+        block_blob_service.get_blob_to_stream(
+            container_name=params["container"],
+            blob_name=params["blob"],
+            stream=output_stream,
         )
 
+        output_stream.seek(0)
+        content = output_stream.readlines()
+        return [
+            line.decode("utf-8") if isinstance(line, six.binary_type) else line
```

Comment:
Just for my education: when does it happen that some lines are `bytes` and others are not within the same stream?

Reply:
You are right. Checking the lines to be `bytes` is unnecessary. The BytesIO stream should always return `bytes`. I will change this